src/Polytopes/Cone.jl

```diff
@@ -0,0 +1,112 @@
+@doc Markdown.doc"""
+    Cone(Rays)
+
+     A polyhedral cone, not necessarily pointed, defined by the positive hull
+     of the rays, Rays.
+"""
+struct Cone #a real polymake polyhedron
+    pm_cone::Polymake.BigObjectAllocated
+end
+function Cone(Rays::Union{Oscar.MatElem,AbstractMatrix})
+    Cone(Polymake.polytope.Cone{Polymake.Rational}(
+        RAYS = matrix_for_polymake(Rays),
+    ))
+end
+
+"""
+    pm_polytope(C::Cone)
+
+Get the underlying polymake `Cone`.
+"""
+pm_cone(C::Cone) = C.pm_cone
+
+struct ConeRayIterator
+    cone::Cone
+end
+
+function Base.iterate(iter::ConeRayIterator, index = 1)
+    rays = pm_cone(iter.cone).RAYS
+    if size(rays, 1) < index
+        return nothing
+    end
+
+    return (rays[index, :], index + 1)
+end
+Base.eltype(::Type{ConeRayIterator}) = Polymake.VectorAllocated{Polymake.Rational}
+Base.length(iter::ConeRayIterator) = n_rays(iter.cone)
+
+"""
+   n_rays(C::Cone)
+
+Returns the number of rays of the cone `C`.
+"""
+n_rays(C::Cone) = size(pm_cone(C).RAYS, 1)
```

Comment:
I think elsewhere we used e.g. `ngens`, and Julia has `ndims`, `nfields`, so I think this should be `nrays` to fit in the pattern.